src/Framework/Framework/Resources/Scripts/state-manager.ts

```diff
@@ -404,3 +404,21 @@ function createWrappedObservable<T>(initialValue: DeepReadonly<T>, typeHint: Typ
     });
     return obs
 }
+
+function compareObjectTypes(currentValue: any, newVal: any): boolean {
+    if (currentValue["$type"] && currentValue["$type"] === newVal["$type"]) {
+        // objects with type must have a same type
+        return true;
+    }
+    else if (!currentValue["$type"] && !newVal["$type"]) {
+        // dynamic objects must have the same properties
+        let currentValueKeys = keys(currentValue);
+        let newValKeys = keys(newVal);
+        if (currentValueKeys.length == newValKeys.length) {
+            currentValueKeys.sort();
+            newValKeys.sort();
+            return currentValueKeys.every((v, i) => newValKeys[i] === v);
+        }
```

Comment:
```suggestion
       return new Set([...currentValueKeys, ...newValKeys]).size == currentValueKeys.length
```
less error prone to indexes, probably less readable... I though the Set class would have some equals method, but no :|